django/contrib/admin/static/admin/js/actions.js

```diff
@@ -142,28 +142,70 @@
+                });
+
+            Array.from(document.getElementById('result_list').tBodies).forEach(
+                function(el) {
+                    el.addEventListener('change', function(event) {
+                        const target = event.target;
+                        if (target.classList.contains('action-select')) {
+                            const checkboxes = affectedCheckboxes(target, shiftPressed);
+                            checker(checkboxes, options, target.checked);
+                            updateCounter(actionCheckboxes, options);
+                            lastChecked = target;
+                        } else {
+                            list_editable_changed = true;
+                        }
+                    });
+                });
+
+            document.querySelector('#changelist-form button[name=index]').addEventListener('click', function(event) {
+                if (list_editable_changed) {
+                    const confirmed = confirm(gettext("You have unsaved changes on individual editable fields. If you run an action, your unsaved changes will be lost."));
+                    if (!confirmed) {
+                        event.preventDefault();
+                    }
+                }
+            });
+        }
+
+
+        if (options.inline) {
+            const handleCheckboxChange = (event) => {
                 const target = event.target;
-                if (target.classList.contains('action-select')) {
+
+                if (!lastChecked || !target.name.endsWith('-DELETE')) {
+                    lastChecked = target;
+                    return;
+                }
+
+                if (lastChecked.name.slice(0, -9) === target.name.slice(0, -9)) {
                     const checkboxes = affectedCheckboxes(target, shiftPressed);
                     checker(checkboxes, options, target.checked);
-                    updateCounter(actionCheckboxes, options);
-                    lastChecked = target;
-                } else {
-                    list_editable_changed = true;
                 }
-            });
-        });
 
-        document.querySelector('#changelist-form button[name=index]').addEventListener('click', function(event) {
-            if (list_editable_changed) {
-                const confirmed = confirm(gettext("You have unsaved changes on individual editable fields. If you run an action, your unsaved changes will be lost."));
-                if (!confirmed) {
-                    event.preventDefault();
+                lastChecked = target;
+            };
+
+            const attachChangeListener = (element) => {
+                if (!element) {
+                    return;
                 }
-            }
-        });
+                element.addEventListener('change', handleCheckboxChange);
+            };
+
+            // Handle tabular inline tables
+            document.querySelectorAll('.tabular_inline_table tbody').forEach(attachChangeListener);
```

Comment:
```suggestion
            document.querySelectorAll('.tabular tbody').forEach(attachChangeListener);
```
Then we can remove adding `class="tabular_inline_table`